modules/yarpc/yarpc.go

```diff
@@ -281,16 +247,17 @@ func newYARPCModule(
 		statsClient: newStatsClient(host.Metrics()),
 		log:         ulog.Logger(context.Background()).With(zap.String("module", host.Name())),
 	}
-	if err := host.Config().Scope("modules").Get(host.Name()).PopulateStruct(&module.config); err != nil {
-		return nil, errs.Wrap(err, "can't read inbounds")
-	}
+
+	val := host.Config().Scope("modules").Get(host.Name()).Value()
 
 	// iterate over inbounds
-	transportsIn, err := prepareInbounds(module.config.Inbounds, host.Name())
+	c, err := cfg.New().LoadConfig(val)
```

Comment:
You'll want to pass in the `map[string]interface{}` for *just* the `yarpc:` key - is that what you're trying to do?